cli/buildengine.ts

```diff
@@ -411,7 +411,8 @@ export function buildDalConst(buildEngine: BuildEngine, mainPkg: pxt.MainPackage
                 enumVal = -1;
             }
 
-            if (inEnum && (m = /^\s*(\w+)\s*(=\s*(.*?))?,?\s*$/.exec(ln))) {
+            m = /^\s*(\w+)\s*(=\s*(.*?))?,?\s*$/.exec(ln);
+            if (inEnum && m) {
```

Comment:
why?

Reply:
no conditional assignment

Reply:
Right.. Well no conditional assignment means you need to take it all out. Let me do that.

Reply:
it did lazy execute the regex so let's bring it back.